tests/reset_workflow.go

```diff
@@ -753,3 +758,70 @@ func (s *FunctionalSuite) testResetWorkflowRangeScheduleToStart(
 	s.NoError(err)
 	s.True(workflowComplete)
 }
+
+func CaNOnceWorkflow(ctx workflow.Context, input string) (string, error) {
+	if err := workflow.Sleep(ctx, time.Millisecond); err != nil {
+		return "", err
+	}
+	if input != "don't CaN" {
+		return input, workflow.NewContinueAsNewError(ctx, CaNOnceWorkflow, "don't CaN")
+	}
+	return input, nil
+}
+
+func (s *FunctionalSuite) TestResetWorkflow_ResetAfterContinueAsNew() {
+	id := "functional-reset-workflow-test"
+	tq := "functional-reset-workflow-test-taskqueue"
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+	defer cancel()
+
+	// get sdkClient cuz idk how to use s.client.StartWorkflowExecution
```

Comment:
I can change this test to use the existing framework of generating events if that's better. I felt more confident in being able to reproduce it from the sdk like some of the other functional test suites do